classes/controller/FrontController.php

```diff
@@ -501,6 +502,15 @@ protected function assignGeneralPurposeVariables()
         ]);
     }
 
+    protected function getTemplateVarDevice(): array
+    {
+        return [
+            'is_mobile' => $this->context->isMobile(),
+            'is_tablet' => $this->context->isTablet(),
+            'type' => $this->context->getDevice(),
```

Comment:
As the `getDevice()` will check the `isMobile` and `isTablet` method in order to return a Context constant, is needed to sent the `is_*` var too ?

Reply:
yeah to avoid something like that in tpl
`{if $type == 1}`

so I suggest 

`{if $is_mobile}`

because it's more readable and avoids putting constants in the tpl

Reply:
Hi, another possibility for lisibility is adding constants do $type == $smarty.const.TYPE_MOBILE